rust/src/push/evaluator.rs

```diff
@@ -198,8 +198,9 @@ impl PushRuleEvaluator {
             let actions = push_rule
                 .actions
                 .iter()
-                // Filter out "dont_notify" actions, as we don't store them.
-                .filter(|a| **a != Action::DontNotify)
+                // Filter out "dont_notify" and "coalesce" actions, as we don't store them
+                // (since they result in no action by the pushers).
+                .filter(|a| **a != Action::DontNotify && **a != Action::Coalesce)
```

Comment:
Pushers only care about `notify` and then tweaks, see

https://github.com/matrix-org/synapse/blob/90af33bda357614deae4acb17888fa5c80e079e3/synapse/push/httppusher.py#L320-L321